java/src/main/java/org/hyperledger/fabric/client/Checkpointer.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+/**
+ * Checkpointer allows update of a checkpoint position after events are successfully processed.
+ */
+public interface Checkpointer extends Checkpoint {
+
+     /**
+      * CheckpointBlock checkpoints block.
+      * @param blockNumber
+      */
+      void  checkpointBlock(long blockNumber) throws Exception;
```

Comment:
Avoid throwing the base Exception class. Be more specific about the type of exception that be thrown whenever practical, so calling code can differentiate between different error conditions from different pieces of code. The error here will be in persisting the checkpoint state, so will be [IOException](https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html).
```suggestion
      void  checkpointBlock(long blockNumber) throws IOException;
```